lumispy/signals/luminescence_transientspec.py

```diff
@@ -21,21 +21,47 @@
 -------------------------------------------------
 """
 
-from hyperspy.signals import Signal2D
+from hyperspy.signals import Signal1D, Signal2D
 from hyperspy._signals.lazy import LazySignal
 
+from lumispy.signals import LumiSpectrum, LumiTransient
 from lumispy.signals.common_luminescence import CommonLumi
 from lumispy.signals.common_transient import CommonTransient
 
 
+class TransientSpectrumCasting(Signal1D, CommonLumi, CommonTransient):
+    """**1D signal class for casting reduced LumiTransientSpectrum to either Luminescence or Transient**"""
```

Comment:
It would be good to explain in more details what it does, because it is an unusual pattern and it is not obvious what it does.

It would be good to add some example showing when it is used (even this will be hidden to the user) to help our future selves and other contributors.